ops/terraform/services/server/server-regression/main.tf

```
@@ -155,3 +157,58 @@ resource "aws_lambda_function" "glue_trigger" {
 
   role = aws_iam_role.glue_trigger.arn
 }
+
+resource "aws_lambda_function" "spice_refresh_trigger" {
+  description      = "Triggers a full SPICE refresh when the server regression glue crawler completes"
+  architectures    = ["x86_64"]
+  function_name    = "bfd-${local.env}-${local.service}-refresh-spice-trigger"
+  handler          = "spice-trigger.lambda_handler"
+  memory_size      = 128
+  package_type     = "Zip"
+  role             = aws_iam_role.spice_refresh_role.arn
+  runtime          = "python3.9"
+  filename         = data.archive_file.spice_trigger.output_path
+  source_code_hash = data.archive_file.spice_trigger.output_base64sha256
+  timeout          = 3
+  environment {
+    variables = {
+      ENV = local.env
+    }
+  }
+}
+
+resource "aws_cloudwatch_event_rule" "spice_refresh_event_rule" {
+  description = "Event to execute a refresh of SPICE server regression ${local.env} dataset"
+  event_pattern = jsonencode(
+    {
+      detail = {
+        crawlerName = [
+          "bfd-insights-bfd-${local.env}-server-regression",
+        ]
+        state = [
+          "Succeeded",
+        ]
+      }
+      detail-type = [
+        "Glue Crawler State Change",
+      ]
+      source = [
+        "aws.glue",
+      ]
+    }
+  )
+  name = "bfd-${local.env}-${local.service}-refresh-spice-event"
```

```suggestion
  name = "${local.spice_trigger_lambda_name}-on-glue-crawler-success"
```

nit: This one's definitely more of a nit, but since this Event Rule is used to invoke the Lambda, I recommend prefixing the name with the name of the Lambda to make the relationship more obvious. Not a blocker for approval, though, just more of a personal preference.